ord_schema/proto/reaction.proto

```diff
@@ -818,32 +810,42 @@ message ReactionWorkup {
 message ReactionOutcome {
   // Reaction time (for flow, equivalent to residence time or spacetime).
   Time reaction_time = 1;
-  // Conversion with respect to the limiting reactant.
+  // Conversion with respect to the sole limiting reactant. If there is
+  // ambiguity in which species the conversion is defined with respect to, the
+  // quantification of leftover starting materials should be defined as the
+  // quantification of multiple products, where the compound identity of those
+  // products matches the starting materials.
   Percentage conversion = 2;
+  // The products observed in this reaction, including product-specific data
+  // defined in submessages. If the quantities of leftover starting materials
+  // are measured, these starting materials may also be defined as product
+  // species.
   repeated ReactionProduct products = 3;
   // Analyses are stored in a map to associate each with a unique key.
-  // The key is cross-referenced in ReactionProduct messages to indicate
-  // which analyses were used to derive which performance values/metrics.
-  // The string used for the key carries no meaning outside of this
-  // cross-referencing.
-  map<string, ReactionAnalysis> analyses = 4;
+  // The key is cross-referenced in ReactionProduct's ProductMeasurement
+  // messages to indicate which analyses were used to derive which performance
+  // values/metrics. The string used for the key carries no meaning outside of
+  // this cross-referencing.
+  map<string, Analysis> analyses = 4;
 }
 
 message ReactionProduct {
-  Compound compound = 1;
+  // The identity of the product compound.
+  repeated CompoundIdentifier identifiers = 1;
```

Comment:
Let's preserve the tags for now.

Reply:
Does this mean (1) avoid using the old numbers, (2) add `reserved` statements for the tag numbers, and/or (3) add `reserved` statements for the field names?

Reply:
Definitely (1). And if you delete a field, please do use `reserved` to keep track of the tag numbers.

Note that this _may_ require new field names for renamed messages. For instance `map<string, Analysis>` may need to be a new field rather than overwriting `map<string, ReactionAnalysis>`.

Reply:
I'm slightly confused as to why it is important to avoid those tag numbers / reserve them if the plan is to dump everything to a pbtxt format where the field names are explicit. But I will push an updated version momentarily

Reply:
It's mostly to be safe and keep the *.pb files working. But that's not a huge deal. Up to you.